src/platforms/hosted/ftdi_swd.c

```diff
@@ -237,7 +238,8 @@ static bool ftdi_swd_seq_in_parity_mpsse(uint32_t *const result, const size_t cl
 	uint8_t data_out[5U] = {0};
 	ftdi_jtag_tdi_tdo_seq(data_out, false, NULL, clock_cycles + 1U);
 	const uint32_t data = read_le4(data_out, 0);
-	uint8_t parity = __builtin_parity(data & ((UINT64_C(1) << clock_cycles) - 1U));
+	const uint64_t bitmask = (UINT64_C(1) << clock_cycles) - 1U;
```

Comment:
Given the check in `ftdi_swd_seq_in_parity` makes it impossible for clock_cycles to exceed 32, this can be made a uint32_t and the `UINT64_C(1)` switched to `1U`. If this causes a build error, cast `clock_cycles` to inhibit that.

Reply:
Turns out this was wrong, we'd put that `UINT64_C()` in because the bit mask uses a 64-bit intermediary - the switch to `uint32_t` is fine, but `1U << 32U` is generated when `clock_cycles` is 32 which has a 64-bit result and invokes undefined behaviour if attempted in 32-bit space. Found while testing on our Tigard.